src/contentmanager.cpp

```diff
@@ -392,30 +392,30 @@ void ContentManager::setSearch(const QString &search)
 QStringList ContentManager::getBookIds()
 {
     kiwix::Filter filter;
-    std::vector<std::string> tags;
+    std::vector<std::string> acceptTags, rejectTags;
     if (m_categoryFilter != "all" && m_categoryFilter != "other") {
-        tags.push_back("_category:"+m_categoryFilter.toStdString());
-        filter.acceptTags(tags);
+        acceptTags.push_back("_category:"+m_categoryFilter.toStdString());
     }
     if (m_categoryFilter == "other") {
         for (auto& category: S_CATEGORIES) {
             if (category.first != "other" && category.first != "all") {
-                tags.push_back("_category:"+category.first.toStdString());
+                rejectTags.push_back("_category:"+category.first.toStdString());
             }
         }
-        filter.rejectTags(tags);
     }
 
     for (auto &contentTypeFilter : m_contentTypeFilters) {
         auto state = contentTypeFilter->checkState();
         auto filter = contentTypeFilter->getName();
         if (state == Qt::PartiallyChecked) {
-            tags.push_back("_" + filter.toStdString() +":yes");
+            acceptTags.push_back("_" + filter.toStdString() +":yes");
         } else if (state == Qt::Checked) {
-            tags.push_back("_" + filter.toStdString() +":no");
+            acceptTags.push_back("_" + filter.toStdString() +":no");
         }
```

Comment:
I wonder why a partially checked state is translated to a "_contenttype:yes" tag and a fully checked state to a "_contenttype:no" tag

Reply:
The button is a 3 states button (`yes`, `no` and `no filtering`).
The 3 states available from Qt are `Checked`, `PartiallyChecked` and `Unchecked`.
We set the button ui here : https://github.com/kiwix/kiwix-desktop/blob/master/src/contenttypefilter.cpp#L8-L12

Reply:
Such mapping of various checked states to filter states is a little confusing. Anyway, your explanation allows to conclude that everything is correct here.